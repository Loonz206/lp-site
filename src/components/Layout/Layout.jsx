import React, { useState, useEffect } from "react";
import PropTypes from "prop-types";
import { Link } from "gatsby";
import Header from "../Header/Header";
import Footer from "../Footer/Footer";
import "./Layout.scss";
import Logo from "../../assets/logo.svg";

const Layout = props => {
  const [state, setState] = useState("hello");
  const handleClick = () => {
    return state === "hello" ? setState("boob") : setState("hello");
  };
  const { children } = props;
  // TODO: Move these links to be dynamic instead and fed into LKP
  const links = [
    {
      name: "about",
      path: "/about",
      id: 0
    },
    {
      name: "services",
      path: "/services",
      id: 1
    },
    {
      name: "contact",
      path: "/contact",
      id: 2
    }
  ];
  useEffect(() => {
    const body = document.querySelector("body");
    if (body && !body.classList.contains("js")) {
      body.classList.add("js");
    }
  }, []);
  return (
    <div className={state === "hello" ? "wrap" : "wrap active"} id="wrap">
      <Header handleClick={handleClick} state={state} links={links} />
      <div className="card-container">
        <Link to="/">
          <Logo className="img-responsive" />
          UX/UI Designer & Front-End Developer
        </Link>
        <br />
        <a href="https://twitter.com/loonz206">@loonz206</a>
      </div>
      <main id="content">{children}</main>
      <Footer />
    </div>
  );
};

export default Layout;

Layout.defaultProps = {
  children: []
};

Layout.propTypes = {
  children: PropTypes.node
};
