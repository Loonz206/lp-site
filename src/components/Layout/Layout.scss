.card-container {
  text-align: center;
  margin: 20px auto;
  clear: both;

  svg {
    max-width: 180px;
    height: 180px;
    border-radius: 50%;
    margin: 0 auto 1em;
    transition: all 2s cubic-bezier(1, -0.53, 0.405, 1.425);

    &:hover,
    &:focus {
      -webkit-transform: rotate(360deg);
      /* Safari */
      transform: rotate(360deg);
      -webkit-filter: grayscale(100%);
      filter: grayscale(100%);
    }

    .center-text {
      text-align: center;
    }
  }
}

// Main
main {
  max-width: 600px;
  margin: auto;
  padding: 20px;
  min-height: 42em;
}
