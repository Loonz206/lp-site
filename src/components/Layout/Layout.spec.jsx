import React from "react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";
import { render, fireEvent } from "@testing-library/react";

import Layout from "./Layout";

describe("Layout", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Layout />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("the header to toggle the UI", () => {
    render(<Layout />);
    // Arrange
    expect(document.querySelector("#wrap")).toHaveClass("wrap");
    // Act
    fireEvent.click(document.querySelector(".menu-link"));
    // Assert
    expect(document.querySelector("#wrap")).toHaveClass("wrap active");
  });
});
