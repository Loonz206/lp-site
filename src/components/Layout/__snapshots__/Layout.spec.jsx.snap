// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  className="wrap"
  id="wrap"
>
  <header>
    <a
      className="sr-only"
      href="#content"
    >
      Skip to Content
    </a>
    <a
      className="menu-link"
      href="#menu"
      onClick={[Function]}
    >
      Menu â˜°
    </a>
    <nav
      id="menu"
      role="navigation"
    >
      <ul>
        <li>
          <a
            className="nav-link"
            href="/about"
          >
            about
          </a>
        </li>
        <li>
          <a
            className="nav-link"
            href="/services"
          >
            services
          </a>
        </li>
        <li>
          <a
            className="nav-link"
            href="/contact"
          >
            contact
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <div
    className="card-container"
  >
    <a
      href="/"
    >
      <test-file-stub
        className="img-responsive"
      />
      UX/UI Designer & Front-End Developer
    </a>
    <br />
    <a
      href="https://twitter.com/loonz206"
    >
      @loonz206
    </a>
  </div>
  <main
    id="content"
  />
  <footer>
    <p>
      Copyright | 2019 - 2022
    </p>
  </footer>
</div>
`;
